pep-0497-externals.txt

```diff
@@ -0,0 +1,359 @@
+
+A common user experience is to attempt the installation of software in a
+basic environment, and for the installation to fail during the compilation
+of a dependency. The user is presented with an error message provided by the
+compiler. Not all users have the ability to understand these messages, and
+a common experience is to spend time searching online for the appopriate
+package which is provided by the vendor.
+
+The base implementation will also only provide information about missing
+headers, libraries or executables. This is a step forward in that the error
+message is provided very early in the install process, rather than potentially
+many minutes in to a large compile job.
+
+A key aspect of the usability of this tool is a mechanism to provide more useful
+information to the user about how to obtain those missing resources.
+The proposal is to provide a layer inside of pip which is capable of providing
+information about appropriate package names. It is surprising that vendors
+do not already provide a good tool for this, given that finding appropriate
+header and library files is a common experience in the general linux ecosystem.
+However, it is also a critical aspect of the user experience, and something
+which package maintainers will need to be able to specify precisely. The
+proposed approach would be for pip to supply the ability to install an
+inference layer, usable by users and vendors alike. Usage would be along
+the lines of
+
+    pip install vendor_magic darwin
+
+This would transform the user messages relating to missing dependencies from
+
+    Could not find libatlas.so
+
+to
+
+    Could not find libatlas.so. Run 'port install scipy +atlas'
+
+
+Discussion / FAQ / Issues
+=========================
+
+This section will be updated to capture incoming feedback and represents a
+snapshot of the current concerns.
+
+
+Specification of files (headers, libs, bins) vs packages
+--------------------------------------------------------
+
+There was a discussion about whether specific dependencies (named headers,
+libraries, binaries etc) should be included, or whether general package names
+(such as may be expected by yum, apt, conda) were appropriate. Due to the
+number of package management tools, plus the changing nature of those, it
+was decided that concentrating on the specific files was more useful than
+using package names. It also provides looser coupling between the specification
+of externals and the package management tools.
+
+Location in setup.py rather than requirements.txt
+-------------------------------------------------
+
+Requirements.txt is scoped to the use for local deployments to specify
+requirements precisely for the relevant local environment. Setup.py is the
+appropriate place to specify the information for the package in general.
+
+In future, there may be some scope for including similar directives within
+requirements.txt, but that is out of scope for this PEP.
```

Comment:
I think because we're extending the definition of 'distribution name' that we need to consider requirements.txt in this spec.